frontend/src/Navigation.ts

```diff
@@ -85,3 +94,17 @@ export default async ({ dynamicNav, currentNamespace }: DynamicNav) => {
       }
     : routes;
 };
+
+export const useNavigation = ({ dynamicNav, currentNamespace }: DynamicNav): Navigation | RouteProps[] => {
+  const [navigation, setNavigation] = React.useState<Navigation | RouteProps[]>();
+  React.useEffect(() => {
+    if (dynamicNav) {
+      // this is just one off for now, but we can start building on this
+      calculateNavigation({ dynamicNav, currentNamespace }).then((data: Navigation | RouteProps[]) => setNavigation(data));
+    }
+  }, [dynamicNav]);
```

Comment:
We need a safeguard here if the component unmounts during the execution of the hook.

```
  React.useEffect(() => {
    let unmounted = false; // new var
    if (dynamicNav) {
      // this is just one off for now, but we can start building on this
      calculateNavigation({ dynamicNav, currentNamespace }).then((data: Navigation | RouteProps[]) => {
        if (unmounted) return; // gate it setting
        setNavigation(data);
      });
    }
    return () => { unmounted = true; }; // unmount flag
  }, [dynamicNav]);
```

Reply:
Good point! I'll use `useRef` as that value is preserved between re-renders.

Reply:
Hmm, you definitely can... just not sure a useRef is needed. 🤔 

Pretty sure the local variable is all you need.